include/aws/common/linked_list.h

```diff
@@ -0,0 +1,111 @@
+#ifndef AWS_LINKED_LIST_H
+#define AWS_LINKED_LIST_H
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+
+#define aws_linked_list_of(type, name) struct name {                                                                   \
+    struct aws_linked_list_node list;                                                                                  \
+    type value;                                                                                                        \
+}                                                                                                                      \
+
+#define aws_container_of(ptr, type, member)  ((type *)ptr)->member
```

Comment:
Actually hold on a sec, this macro is still broken. What if the list isn't the first member of whatever it's in?

Reply:
The normal definition is something like:

```
#define container_of(ptr, container_type, container_member) \
  ((container_type *)((uint8_t *)ptr - (ptrdiff_t)&((container_type *)NULL)->container_member)
```